src/comms/views.py

```diff
@@ -188,6 +195,58 @@ def news_list(request, tag=None):
     context = {
         'news_items': news_items,
         'tag': tag,
+        'all_tags': all_tags,
+    }
+
+    return render(request, template, context)
+
+
+def press_overview(request):
+    """
+    Gets NewsItem lists by tag or by the content type of the
+    object (e.g. press, journal), in the order they are passed in the URL.
+    Expects GET request query parameters such as in this example:
+    /news/press_overview/?tags__text=Announcement&tags__text=Blog&content_type__model=journal
+    :param request: HttpRequest object
+    :return: HttpResponse object
+    """
+
+    all_tags = models.Tag.objects.all().annotate(
+        Count('tags')
+    ).order_by('-tags__count', 'text')
+
+    tag_texts = [tag.text for tag in all_tags]
+    tag_sets = {}
+    object_sets = {}
+    for key in request.GET:
+        if key == 'tags__text':
```

Comment:
This might just be me but it feels odd to expose the django notation in the GET parameters. Could this just be 'tags' and below 'type'?

Reply:
I think I did this to be more explicit and readable because I was dealing with bugs arising from the ambiguity of "tags"--is it an object, a string, etc.